codegen/templates/workflow.tmpl

```diff
@@ -168,40 +173,70 @@ func (w {{$workflowStruct}}) Handle(
 		}
 	}
 
+	//when maxRetry is 0, timeout per client level is used & one attempt is made, and timoutPerAttempt is not used
+	var timeoutAndRetryOps = zanzibar.TimeoutAndRetryOptions{}
+	var maxRetry int = 0
+	var timeoutPerAttempt time.Duration
+	var backOffTimeAcrossRetries time.Duration
+	var overAllTimeout time.Duration
+
+	//when endpoint level timeout information provided, override it with client level config
+	if w.defaultDeps.Config.ContainsKey("endpoints.{{$handleIdDotEndpointId}}.timeoutPerAttempt") {
+		scaleFactor := w.defaultDeps.Config.MustGetFloat("endpoints.{{$handleIdDotEndpointId}}.scaleFactor")
+		maxRetry = int(w.defaultDeps.Config.MustGetInt("endpoints.{{$handleIdDotEndpointId}}.retryCount"))
+
+		backOffTimeAcrossRetriesConf := w.defaultDeps.Config.MustGetInt("endpoints.{{$handleIdDotEndpointId}}.backOffTimeAcrossRetries")
+		timeoutPerAttemptConf := w.defaultDeps.Config.MustGetInt("endpoints.{{$handleIdDotEndpointId}}.timeoutPerAttempt")
+
+		//add a scale factor to timeoutPerAttempt
+		timeoutPerAttemptConf = int64(math.Ceil(float64(timeoutPerAttemptConf) * scaleFactor))
+
+		//convert timeoutPerAttemptConf & backOffTimeAcrossRetries to Duration
+		timeoutPerAttempt = time.Duration(timeoutPerAttemptConf) * time.Millisecond
+		backOffTimeAcrossRetries = time.Duration(backOffTimeAcrossRetriesConf) * time.Millisecond
+
+		overAllTimeout = time.Duration((timeoutPerAttemptConf + backOffTimeAcrossRetriesConf) * (int64(maxRetry-1)) + timeoutPerAttemptConf) * time.Millisecond
```

Comment:
We don't need to have this logic in `tmpl` This should be sourced from the runtime. tmpl should only be the invoker or contain branching at max. This makes our template bloated and increases code gen size.